frontend/src/concepts/pipelines/content/artifacts/charts/ConfusionMatrix.tsx

```diff
@@ -0,0 +1,185 @@
+import React from 'react';
+import { Text } from '@patternfly/react-core';
+
+export type ConfusionMatrixInput = {
+  annotationSpecs: {
+    displayName: string;
+  }[];
+  rows: { row: number[] }[];
+};
+
+export interface ConfusionMatrixConfig {
+  data: number[][];
+  labels: string[];
+}
+
+export function buildConfusionMatrixConfig(
+  confusionMatrix: ConfusionMatrixInput,
+): ConfusionMatrixConfig {
+  return {
+    labels: confusionMatrix.annotationSpecs.map((annotation) => annotation.displayName),
+    data: confusionMatrix.rows.map((x) => x.row),
+  };
+}
+
+type ConfusionMatrixProps = {
+  config: ConfusionMatrixConfig;
+  size?: number;
+};
+
+const ConfusionMatrix: React.FC<ConfusionMatrixProps> = ({
+  config: { data, labels },
+  size = 100,
+}) => {
+  const max = Math.max(...data.flat());
+
+  // Function to get color based on the cell value
+  const getColor = (value: number) => {
+    const opacity = value / max; // Normalize the value to get opacity
+    return `rgba(41, 121, 255, ${opacity})`; // Use blue color with calculated opacity
+  };
+
+  // Determine the size for all cells, including labels
+  const cellSize = `${size}px`;
+
+  // Generate the gradient for the legend
+  const gradientLegend = `linear-gradient(to bottom, rgba(41, 121, 255, 1) 0%, rgba(41, 121, 255, 0) 100%)`;
+
+  return (
+    <div
+      style={{
+        display: 'flex',
```

Comment:
Can use `Flex` PF component instead of this.

Reply:
Yeah probably, this component is a mess bc it has to override many styles, i will revisit this and see what i ca do to reuse patternfly. But some PF layout components add extra styles that i cant have in this weird usage

Reply:
the Flex broke it so i just put everything into a scss file to clean it up